lib/osut/utils.rb

```diff
@@ -3454,6 +3548,15 @@ def fits?(p1 = nil, p2 = nil, entirely = false)
 
     p1.each { |p0| return false unless pointWithinPolygon?(p0, p2) }
 
+    # Although p2 points may lie ALONG p1, none may lie entirely WITHIN p1.
+    p2.each { |p0| return false if pointWithinPolygon?(p0, p1, true) }
+
+    # p1 segment mid-points must not lie OUTSIDE of p2.
+    getSegments(p1).each do |sg|
+      mp = midpoint(sg.first, sg.last)
+      return false unless pointWithinPolygon?(mp, p2)
+    end
```

Comment:
This is an expensive add-on, but essential when dealing with convoluted surfaces, namely holding multiple sequential concavities, like C-shaped or G-shaped corridors. Want to speed things up? Break up such a surface into multiple convex ones.